sharding/contracts/validator_manager_test.go

```diff
@@ -0,0 +1,62 @@
+package contracts
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var (
+	key, _ = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+	addr   = crypto.PubkeyToAddress(key.PublicKey)
+)
+
+func TestContractCreation(t *testing.T) {
```

Comment:
Can you be more specific? This is testing a lot of things other than just deploying the contract. 

Please split up these into smaller tests where each test method has a suite of test conditions and expected output.